### ISSUE TITLE
 [DioErrorType.DEFAULT]: SocketException: OS Error: Connection refused,

### ISSUE THREAD

**Issue by codeThe0** [flutterchina/dio#546]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by JJwow** [comment]
me too! but just in Android

**Comment by ganisback** [comment]
+1

**Comment by kleon1024** [comment]
+1

**Comment by visual-velocity** [comment]
I am also getting this error, how can i handle it ?

**Comment by orestesgaolin** [comment]
Also getting the error. Usually happens after 3-4 successful requests, and then immediately fails with:

```
 SocketException: OS Error: Connection refused, errno = 61, address = web.address.here, port = 58931
```

**Comment by Vitaliy-Svinchyak** [comment]
The same. Making a request to the  http://localhost:5002. Always getting such error `Connection refused, errno = 111, address = localhost, port = 47026` with random port

**Comment by Vitaliy-Svinchyak** [comment]
Problem: requests to localhost are not working because on android localhost is the phone itself
Solution: run `adb reverse tcp:5002 tcp:5002` to proxy it
Thanks: https://github.com/flutterchina/dio/issues/76#issuecomment-430011020

**Comment by t-kietzmann** [comment]
I'm currently facing this with the Apple App Review Team. The app trys to download some files from firebase storage and this error is thrown (have a screenshot from the Review-Team):

Socket Exception: Connection refused (OS Error: Connection refused, errno = 61), address = firebasestorage.googleapis.com, port = 50323

I can not reproduce this error/behaviour - neither in the debug mode nor release - also no TestFlight-Tester has this problem.  
Additionally i deleted all storage rules.

**Comment by ChrisMICDUP** [comment]
@t-kietzmann you may be seeing this issue because a corporate firewall is blocking access to firebasestorage.googleapis.com. We have instances of this in our crashlytics logs from a user at their work downloading a static image stored in a GCP bucket, that they have no problem accessing when on cell data or from their home.

I'm working on what will be the most cost effective way to proxy the call. (it will be probably be trivial...)

**Comment by t-kietzmann** [comment]
yeah figured it out... seems that the review works much easier when the employee is in home office :D

**Comment by ChrisMICDUP** [comment]
Of course the real question is how many companies do this...
